### PR TITLE
More complex model intruduced

### PR DESCRIPTION
**Pull request by SlyCright**

### CODE REVIEW

File: src/main/java/educationalproject/programmingstuff/model/CommodityItem.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@NoArgsConstructor
+@Table(name = "StoreHouse")
+@Builder
+public class CommodityItem {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @OneToOne(cascade = CascadeType.MERGE, optional = false)
```
Comment (philkuzmin) [inline src/main/java/educationalproject/programmingstuff/model/CommodityItem.java:23]: why merge? just askin'...

File: src/main/java/educationalproject/programmingstuff/servicies/CommodityItemServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+@Service
+@RequiredArgsConstructor
+@Validated
+public class CommodityItemServiceImpl implements CommodityItemService {
+
+    @Autowired
+    private CommodityItemRepository commodityItemRepository;
+
+    @Autowired
```
Comment (philkuzmin) [inline src/main/java/educationalproject/programmingstuff/servicies/CommodityItemServiceImpl.java]: annotations not needed, but make these fields final

File: src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+@Value
+@Builder
+public class CommodityItemResponseDto {
+
+    @NotNull(message = "Id can't be null")
+    long id;
+
+    @NotNull(message = "Item can't be null")
+    Item item; // TODO: 12.04.2022 Here should be DTO not Entity
```
Comment (philkuzmin) [inline src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java]: yes, DTO can't contain links to entities, only other DTOs

File: src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+@Value
+@Builder
+public class CommodityItemResponseDto {
+
+    @NotNull(message = "Id can't be null")
+    long id;
```
Comment (philkuzmin) [inline src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java]: if it's a primitive type, then @NotNull is useless

File: src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Spy
+    CommodityItemResponseMapper commodityItemResponseMapper = Mappers.getMapper(CommodityItemResponseMapper.class);
+
+    @Spy
+    @InjectMocks
+    CommodityItemServiceImpl commodityItemService;
+
+    @Test
+    void givenInvocationOfGetAllCommodityItems_whenInvoked_whenSucceed() {
```
Comment (philkuzmin) [inline src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java]: givenCommodityItems_whenGetAllCommodityItems_thenSuccess

File: src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        //Then
+        assertThat(resultCommodities.size()).isEqualTo(expectedCommodities.size());
+
+        for (CommodityItemResponseDto result: resultCommodities) {
+
+            int index=resultCommodities.indexOf(result);
+            CommodityItem expect = expectedCommodities.get(index);
+            assertThat(result.getItem()).isEqualTo(expect.getItem());
+            assertThat(result.getQuantity()).isEqualTo(expect.getQuantity());
+        }
```
Comment (philkuzmin) [inline src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java]: can you just convert all expected entities to dtos and then assert them in one string?

File: src/main/java/educationalproject/programmingstuff/model/CommodityItem.java (hunk 1 above)
Comment (SlyCright) [reply src/main/java/educationalproject/programmingstuff/model/CommodityItem.java:23]: "Merge" works. "All" doesn't. Got error on data prep phase

File: src/main/java/educationalproject/programmingstuff/servicies/CommodityItemServiceImpl.java (hunk 2 above)
Comment (SlyCright) [reply src/main/java/educationalproject/programmingstuff/servicies/CommodityItemServiceImpl.java]: Looks like the same issue as "Order1" was. Tests didn't start without "Autowired" annotations. Now starts. Deleted.

File: src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java (hunk 3 above)
Comment (SlyCright) [reply src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java]: Yep, I just hope someone shows me typical solution for that, so i don't need to invent custom one :)

File: src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java (hunk 4 above)
Comment (SlyCright) [reply src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java]: Done

File: src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java (hunk 5 above)
Comment (SlyCright) [reply src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java]: Done

File: src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java (hunk 6 above)
Comment (SlyCright) [reply src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java]: To discuss. As I assume the way of unit tests works that solution would be wrong.

File: src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java (hunk 3 above)
Comment (SlyCright) [reply src/main/java/educationalproject/programmingstuff/servicies/dto/CommodityItemResponseDto.java]: Done

File: src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java (hunk 6 above)
Comment (SlyCright) [reply src/test/java/educationalproject/programmingstuff/tests_units/CommodityItemServiceImplTest.java]: Done

**Pull request merged**